### ISSUE TITLE
MODX 3.0.0 Error Log: warnings when upload pictures via media browser

### ISSUE THREAD

**Issue by jolichter** [modxcms/revolution#16113]
### Summary
Then upload pictures via media browser get a lot of phpthumb.class.php and modPhpThumb.php warnings

### Step to reproduce
 - new MODX 3.0.0 installation on localhost without having installed any extras
 - upload pictures via media browser 

### Observed behavior
- no errors with MODX 2.8.3 (PHP 7.4.16) on the same Linux System about two months ago
- therefore I am not sure if it is a MODX 3.0.0 or (and) a PHP 8.1.3 problem

### Expected behavior
- no warnings in Error Log after uploading pictures ;-)

### Environment
- MODX Revolution 3.0.0-pl (traditional)
- System Linux 5.15.28-1-MANJARO (based on Arch)
- Apache/2.4.52 (Unix) PHP/8.1.3 
- mysql: 10.7.3-MariaDB (database charset: utf8mb4)

### Error Log
`
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_enabled
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_directory
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$allow_local_http_src
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_directory_depth
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 1522) PHP warning: file_get_contents(/home/modx/public_html/modx/core/cache/phpthumb/phpThumbCacheIMversion.txt): Failed to open stream: Datei oder Verzeichnis nicht gefunden
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 1454) PHP warning: file_get_contents(/home/modx/public_html/modx/core/cache/phpthumb/phpThumbCacheIMcommandlineBase.txt): Failed to open stream: Datei oder Verzeichnis nicht gefunden
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 1432) PHP warning: file_get_contents(/home/modx/public_html/modx/core/cache/phpthumb/phpThumbCacheIMwhichConvert.txt): Failed to open stream: Datei oder Verzeichnis nicht gefunden
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_enabled
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_directory
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$allow_local_http_src
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_directory_depth
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "sia"
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "down"
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 337) PHP warning: Undefined array key "down"
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 339) PHP warning: Undefined array key "down"
[2022-04-02 20:23:18] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 305) PHP warning: Undefined array key "HTTP_IF_MODIFIED_SINCE"
[2022-04-02 20:23:19] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "sia"
[2022-04-02 20:23:19] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "down"
[2022-04-02 20:23:19] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 337) PHP warning: Undefined array key "down"
[2022-04-02 20:23:19] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 339) PHP warning: Undefined array key "down"
[2022-04-02 20:23:19] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 305) PHP warning: Undefined array key "HTTP_IF_MODIFIED_SINCE"
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_enabled
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_source_directory
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$allow_local_http_src
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/vendor/james-heinrich/phpthumb/phpthumb.class.php : 338) PHP warning: Undefined property: MODX\Revolution\modPhpThumb::$cache_directory_depth
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "sia"
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 335) PHP warning: Undefined array key "down"
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 337) PHP warning: Undefined array key "down"
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 339) PHP warning: Undefined array key "down"
[2022-04-02 20:23:27] (ERROR @ /home/modx/public_html/modx/core/src/Revolution/modPhpThumb.php : 305) PHP warning: Undefined array key "HTTP_IF_MODIFIED_SINCE"
`

**Comment by PuddingtonBeer** [comment]
I'm seeing the same "undefined property" x 5 running 2.8.4 on PHP 8.0. Could it be something that was changed in 2.8.4/3.0.0 at the same time?

**Comment by Jako** [comment]
These errors are 'just' PHP warnings and can be disabled, when `error_reporting` is not including `E_WARNING`.

**Comment by PuddingtonBeer** [comment]
Understood, but it only caught my attention because after updating from 2.8.3 -> 2.8.4 my error log started growing at about 2.5MB/day. I've since noticed that this particular site was using phpThumbOf, which I've just replaced with pThumb, and now the error log remains empty so far. pThumb is only a little less old, but perhaps 2.8.4 updated phpThumb in such a way that the older package was no longer a tidy match.
